### ISSUE TITLE
-I<include> bug in batch

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#6086]
**Description of the problem**

I cannot get [animation 12](https://docs.generic-mapping-tools.org/dev/animations/anim12.html) to run on any gmt versions. The issue seems to be that the include file variables are not available for all of the batch scripts. I modified to example from the batch docs for a simpler example that also fails.

**Full script that generated the error**

```
cat << EOF > inc.sh
REGION=EU
EOF
cat << EOF > pre.sh
echo "gmt coast -E=${REGION}+l > countries.txt"
gmt begin
    gmt coast -E=${REGION}+l > countries.txt
gmt end
EOF
cat << EOF > main.sh
gmt begin \${BATCH_NAME} pdf
    gmt coast -R\${BATCH_WORD0}+r2 -JQ10c -Glightgray -Slightblue -B -B+t"\${BATCH_WORD1}" -E\${BATCH_WORD0}+gred+p0.5p
    echo \${BATCH_WORD0} | gmt text -F+f16p+jTL+cTL -Gwhite -W1p
gmt end
EOF
cat << EOF > post.sh
gs -dQUIET -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE=\${BATCH_PREFIX}.pdf -dBATCH \${BATCH_PREFIX}_*.pdf
rm -f \${BATCH_PREFIX}_*.pdf
EOF
gmt batch -Iinc.sh main.sh -Sbpre.sh -Sfpost.sh -Tcountries.txt+w"\t" -Ncountries -V -W -Zs
```

**Full error message**

```
batch [INFORMATION]: Create preflight script batch_preflight.sh and execute it
gmt coast -E=+l > countries.txt
batch [INFORMATION]: Reading Data Table from File countries.txt
batch [WARNING]: File countries.txt is empty!
ERROR: Caught signal number 11 (Segmentation fault: 11) at
0   libgmt.6.4.0.dylib                  0x0000000100f183ae GMT_batch + 11774
```

**System information**

* Operating system: macOS
* GMT version (`gmt --version`): 6.4.0_793bc42_2021.12.02

**Comment by PaulWessel** [comment]
Will have a look - finally made it into the office at UHM,..

**Comment by PaulWessel** [comment]
The pre.sh script has this bad command

`gmt coast -E=+l > countries.txt`

since ${REGION} is not escaped.  I think you want it to be

`gmt coast -E=${REGION}+l > countries.txt`

which means the original script needs to escape those dollar signs or use 'EOF'.

**Comment by meghanrjones** [comment]
I still get lots of errors after adding the proper 'EOF':
```
cat << 'EOF' > inc.sh
REGION=EU
EOF
cat << 'EOF' > pre.sh
echo "gmt coast -E=${REGION}+l > countries.txt"
gmt begin
    gmt coast -E=${REGION}+l > countries.txt
gmt end
EOF
cat << 'EOF' > main.sh
gmt begin \${BATCH_NAME} pdf
    gmt coast -R\${BATCH_WORD0}+r2 -JQ10c -Glightgray -Slightblue -B -B+t"\${BATCH_WORD1}" -E\${BATCH_WORD0}+gred+p0.5p
    echo \${BATCH_WORD0} | gmt text -F+f16p+jTL+cTL -Gwhite -W1p
gmt end
EOF
cat << 'EOF' > post.sh
gs -dQUIET -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE=\${BATCH_PREFIX}.pdf -dBATCH \${BATCH_PREFIX}_*.pdf
rm -f \${BATCH_PREFIX}_*.pdf
EOF
gmt batch -Iinc.sh main.sh -Sbpre.sh -Sfpost.sh -Tcountries.txt+w"\t" -Ncountries -V -W
```

**Comment by PaulWessel** [comment]
Ran without a hitch for me.  Perhaps you will need to run with -Qs and then manually run one of the jobs to see what is goihg on?

**Comment by meghanrjones** [comment]
I fixed the simple example by removing the mix of escape sequences and 'EOF's. Does [anim12](https://docs.generic-mapping-tools.org/dev/animations/anim12.html) also run without a hitch for you?
```
#!/usr/bin/env bash
#
# Blending the NASA day and night views from the Blue and Black Marble mosaic
# images using a day-night mask that goes through a full 24-hour cycle.  In addition,
# we adjust the colors using the intensities derived from the slopes of the earth
# relief grid.  We spin around at 24 frames per second where each frame advances by 1 minute.
# Because we are not plotting anything, just manipulating images via grids, we must instead
# create the frames and the final movie via the batch module.
#
# DEM:    @earth_relief_04m
# Images: @earth_day_HD.tif @earth_night_HD.tif from the GMT cache server
# 
# The finished movie is available in our YouTube channel as well:
# https://youtu.be/X8TojLs0NYk
# The movie took ~4 minutes to render on a 24-core MacPro 2013.
# To make a UHD (4k) movie, comment/uncomment the SIZE, INC and FILT settings
cat << 'EOF' > inc.sh
SIZE=HD
INC=11.25
FILT=21
# SIZE=UHD
# INC=5.625
# FILT=10.5
EOF
# 1. Create preflight script to build data files needed in the loop
cat << 'EOF' > pre.sh
gmt begin
  # Set view time for June 20-21, 2020 with steps of 1 minute
  gmt math -T2020-06-20T/2020-06-21T/1 -o0 --TIME_UNIT=m  T = times.txt
  # Create a ${SIZE} DEM at ${INC}x${INC} arc minutes to yield ${SIZE} dimensions
  gmt grdfilter @earth_relief_04m -Fg${FILT} -I${INC}m -D1 -r -Gtopo_${SIZE}.grd
  # Create an intensity grid based on the ${SIZE} DEM so we can see structures in the oceans
  gmt grdgradient topo_${SIZE}.grd -Nt0.5 -A45 -Gintens_${SIZE}.grd
  # Make sure our remote files have been downloaded
  gmt which -Ga @earth_day_${SIZE}.tif @earth_night_${SIZE}.tif
gmt end
EOF
# 2. Set up main script
cat << 'EOF' > main.sh
gmt begin
  # Let HSV minimum value go to zero and faint map border
  gmt set COLOR_HSV_MIN_V 0 MAP_FRAME_PEN=faint
  # Make global grid with a smooth 2-degree day/night transition for this time.
  gmt grdmath -Rd -I${INC}m -r $(gmt solar -C -o0:1 -I+d${BATCH_COL0}) 2 DAYNIGHT = daynight_${SIZE}.grd
  # Blend the day and night Earth images using the weights, so that when w is 1
  # we get the daytime view, and then adjust colors based on the intensity.
  gmt grdmix @earth_day_${SIZE}.tif @earth_night_${SIZE}.tif -Wdaynight_${SIZE}.grd -Iintens_${SIZE}.grd -G${BATCH_NAME}.png -Ve
gmt end
EOF
# 3. Create postflight script to build movie from the images
cat << 'EOF' > post.sh
gmt begin
  prec=$(gmt math -Q ${BATCH_NJOBS} LOG10 CEIL =)
  ffmpeg -loglevel warning  -f image2 -framerate 24 -y -i "${BATCH_PREFIX}_%0${prec}d.png" -vcodec libx264 -pix_fmt yuv420p ${BATCH_PREFIX}_${SIZE}.mp4
  rm -f ${BATCH_PREFIX}_*.png
gmt end
EOF
# 4. Run the batch, requesting a fade in/out via white
gmt batch main.sh -Iinc.sh -Sbpre.sh -Sfpost.sh -Ttimes.txt -Nanim12 -V -W -Zs
```

**Comment by PaulWessel** [comment]
I will run it in a few minutes - using a lot of cores on something else right now..

**Comment by PaulWessel** [comment]
OK, I see there are issues with the include script. Will check.

**Comment by PaulWessel** [comment]
Fix is coming after some tests complete.

**Comment by meghanrjones** [comment]
Fixed by #6088
